sdk-core/src/test/java/care/data4life/sdk/auth/UserServiceTest.kt

```diff
@@ -0,0 +1,251 @@
+    }
+
+    @Test
+    fun `getSessionToken should Return True`() {
+        //given
+        every { authService.refreshAccessToken(USER_ALIAS) } returns AUTH_TOKEN
+        // when
+        val testSubscriber = userService.getSessionToken(USER_ALIAS)
+            .test()
+
+        // then
+        testSubscriber
+            .assertNoErrors()
+            .assertValue(AUTH_TOKEN)
+    }
+
+    @Ignore
+    @Test
+    fun `getSessionToken should Throws Error`() {
+        //given
+        every { authService.refreshAccessToken(USER_ALIAS) } returns mockk<String>()
+        // when
+        val testSubscriber = userService.getSessionToken(USER_ALIAS)
+            .test()
+            .await()
+
+        // then
+        testSubscriber
+            .assertNoValues()
+    }
+
+    @Test
+    fun `Given VersionList is fetched, when the call fails return true`() {
+        //given
+        val error = RuntimeException("error")
+        val response: Single<VersionList> = Single.fromCallable { throw error }
+        mockkObject(Log)
+        every { apiService.fetchVersionInfo() } returns response
+        //when
+        val testSubscriber = userService.getVersionInfo("1.9.0")
+            ?.test()
+            ?.await()
+        //then
+        testSubscriber?.assertValue(true)
+        verify(exactly = 1) { Log.error(error, "Version not supported") }
+    }
+
+    @Test
+    fun `Given VersionList is fetched, when version supported return true`() {
+        val versions: VersionList = mockk()
+        val response: Single<VersionList> = Single.just(versions)
+        val currentVersion = "1.9.0"
+        every { apiService.fetchVersionInfo() } returns response
+        every { versions.isSupported(currentVersion) } returns true
+        //when
+        val testSubscriber = userService.getVersionInfo("1.9.0")
+            ?.test()
+        //then
+        testSubscriber?.assertValue(true)
+            ?.assertNoErrors()
+
+    }
+
+    @Test
+    fun `logout should Return Success`() {
+        //given
+        every { apiService.logout(USER_ALIAS) } returns Completable.complete()
+        every { storage.clear() } just Runs
+        // when
+        val testSubscriber = userService.logout()
+            .test()
+            .await()
+
+        // then
+        testSubscriber
+            .assertNoErrors()
+            .assertComplete()
+        verify { storage.clear() }
+    }
+
+    @Test
+    fun `logout should Return Exception`() {
+        //given
+        every { apiService.logout(USER_ALIAS) } returns Completable.error(Exception())
+        // when
+        val testSubscriber = userService.logout()
+            .test()
+            .await()
+
+        // then
+        testSubscriber
+            .assertError(Exception::class.java)
+            .assertNotComplete()
+        verify { storage wasNot Called }
+    }
+
+    @Test
+    fun `finish Login should Return Boolean Success`() {
+        //given
+        val userInfo = mockk<UserInfo>()
```

Comment:
Could you please make here a usage of variables? So this can get rid of the `any()` parameter?

Reply:
why is it important to not use any? in this test I only want to check finishLogin method.

Reply:
Much better now, thanks!